tiledb/sm/query/readers/filtered_data.h

```diff
@@ -604,6 +605,9 @@ class FilteredData {
   /** Memory tracker for the filtered data. */
   shared_ptr<MemoryTracker> memory_tracker_;
 
+  /** Resources used to perform operations. */
+  ContextResources& resources_;
```

Comment:
Resource members should always be the first member variable. 

Doing this will forestall a number of order-of-initialization problems in the future. In particular, they should go before memory tracking member variables.